### PR TITLE
Fix Effector.reset() crash on numpy joint_state

### PR DESCRIPTION
**Pull request by syntactic**
`Effector.reset()` and `_parse_initial_joint_state()` assumed `joint_state` was always a `torch.Tensor`, calling `.cpu().detach().numpy()` on it unconditionally. This crashes with an `AttributeError` whenever `joint_state` is a numpy array, which it always is when it comes from `Environment`'s `q_init` (stored as `np.array(...)` in `Environment.__init__`, never a tensor).

Fix: guard both sites with `torch.is_tensor(joint_state)`, matching the pattern `draw_fixed_states` already uses elsewhere in the same file. No behavior change for the existing tensor path.

### CODE REVIEW

**Comment by OlivierCodol** [comment]
if you `MotorNet/tests/` and run `python3 -m pytest .`, can you report the output to confirm this all works? Glancing at the code this looks fine but always good to send unit tests. Thanks

**Comment by syntactic** [comment]
Here's the test output:

```
============================= test session starts ==============================
platform darwin -- Python 3.13.7, pytest-9.0.3, pluggy-1.6.0
rootdir: /Users/syntactic/CodeSandbox/Python/MotorNet
configfile: pyproject.toml
testpaths: tests
plugins: anyio-4.14.2
collected 330 items

tests/test_construction.py ...........................                   [  8%]
tests/test_effector.py ................................................. [ 23%]
....................................                                     [ 33%]
tests/test_environment.py .............................................. [ 47%]
...............................                                          [ 57%]
tests/test_muscle.py ................................................... [ 72%]
.............................                                            [ 81%]
tests/test_simulation.py ...............................                 [ 90%]
tests/test_skeleton.py ..............................                    [100%]

============================= 330 passed in 5.07s ==============================
```

**Review by OlivierCodol — APPROVED** [review]
